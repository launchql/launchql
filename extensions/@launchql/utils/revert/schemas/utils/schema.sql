-- Revert schemas/utils/schema from pg

BEGIN;

DROP SCHEMA utils;

COMMIT;
