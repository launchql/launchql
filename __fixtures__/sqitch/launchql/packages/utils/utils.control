# utils extension
comment = 'skitch project'
default_version = '0.0.1'
module_pathname = '$libdir/utils'
requires = 'plpgsql,uuid-ossp,totp,pg-verify'
relocatable = false
superuser = false
