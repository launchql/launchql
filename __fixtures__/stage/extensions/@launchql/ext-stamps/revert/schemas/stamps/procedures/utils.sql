-- Revert schemas/stamps/procedures/utils from pg

BEGIN;

DROP FUNCTION stamps.utils;

COMMIT;
