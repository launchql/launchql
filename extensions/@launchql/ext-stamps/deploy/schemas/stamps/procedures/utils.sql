-- Deploy schemas/stamps/procedures/utils to pg

-- requires: schemas/stamps/schema

BEGIN;

CREATE FUNCTION stamps.peoplestamps()
RETURNS TRIGGER AS $$
BEGIN
    IF TG_OP = 'INSERT' THEN
      NEW.created_by = jwt_public.current_user_id();
      NEW.updated_by = jwt_public.current_user_id();
    ELSIF TG_OP = 'UPDATE' THEN
      NEW.created_by = OLD.created_by;
      NEW.updated_by = jwt_public.current_user_id();
    END IF;
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';

CREATE FUNCTION stamps.timestamps()
RETURNS TRIGGER AS $$
BEGIN
    IF TG_OP = 'INSERT' THEN
      NEW.created_at = NOW();
      NEW.updated_at = NOW();
    ELSIF TG_OP = 'UPDATE' THEN
      NEW.created_at = OLD.created_at;
      NEW.updated_at = NOW();
    END IF;
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';

COMMIT;
