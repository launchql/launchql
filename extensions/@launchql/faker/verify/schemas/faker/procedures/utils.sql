-- Verify schemas/faker/procedures/utils  on pg

BEGIN;

SELECT verify_function ('faker.utils');

ROLLBACK;
