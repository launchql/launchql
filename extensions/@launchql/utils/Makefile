EXTENSION = launchql-utils
DATA = sql/launchql-utils--0.4.6.sql

PG_CONFIG = pg_config
PGXS := $(shell $(PG_CONFIG) --pgxs)
include $(PGXS)
  