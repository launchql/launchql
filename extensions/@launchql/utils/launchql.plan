%syntax-version=1.0.0
%project=launchql-utils
%uri=launchql-utils
  
schemas/utils/schema 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/schema
schemas/utils/procedures/mask_pad [schemas/utils/schema] 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/procedures/mask_pad
schemas/utils/procedures/throw [schemas/utils/schema] 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/procedures/throw
schemas/utils/procedures/ensure_singleton [schemas/utils/schema] 2021-04-25T11:50:15Z Dan Lynch <dlynch@Dans-MBP-3> # add schemas/utils/procedures/ensure_singleton
