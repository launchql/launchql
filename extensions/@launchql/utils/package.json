{
  "_from": "@launchql/utils@^0.4.6",
  "_id": "@launchql/utils@0.4.6",
  "_inBundle": false,
  "_integrity": "sha512-1ZB6FYCWUCr0casWo7m8Xr74IGpwVSlaTXGnVhslXoLAaU/nwIr39khzRRkHKIrjJNG4VoWfMTymZLfDAQDj1Q==",
  "_location": "/@launchql/utils",
  "_phantomChildren": {},
  "_requested": {
    "type": "range",
    "registry": true,
    "raw": "@launchql/utils@^0.4.6",
    "name": "@launchql/utils",
    "escapedName": "@launchql%2futils",
    "scope": "@launchql",
    "rawSpec": "^0.4.6",
    "saveSpec": null,
    "fetchSpec": "^0.4.6"
  },
  "_requiredBy": [
    "/@pyramation/launchql-meta-schema"
  ],
  "_resolved": "https://registry.npmjs.org/@launchql/utils/-/utils-0.4.6.tgz",
  "_shasum": "4d30448b4bc2efa4891add0b9d393c66ad181333",
  "_spec": "@launchql/utils@^0.4.6",
  "_where": "/private/var/folders/kn/177lldss4n9fwn0gh5q5wfn00000gn/T/ny4sag54vjfki5oa3m947/extensions/node_modules/@pyramation/launchql-meta-schema",
  "author": {
    "name": "Dan Lynch",
    "email": "pyramation@gmail.com"
  },
  "bundleDependencies": false,
  "deprecated": false,
  "description": "utils",
  "devDependencies": {
    "@babel/cli": "7.10.4",
    "@babel/core": "7.10.4",
    "@babel/plugin-proposal-class-properties": "7.10.4",
    "@babel/plugin-proposal-export-default-from": "7.10.4",
    "@babel/plugin-proposal-object-rest-spread": "7.10.4",
    "@babel/plugin-transform-runtime": "7.10.4",
    "@babel/preset-env": "7.10.4",
    "@babel/runtime": "^7.4.2",
    "@launchql/cli": "1.3.5",
    "@launchql/db-testing": "latest",
    "babel-eslint": "10.1.0",
    "babel-jest": "26.1.0",
    "eslint": "^7.3.1",
    "eslint-config-prettier": "^6.10.0",
    "eslint-plugin-prettier": "^3.1.2",
    "jest": "26.1.0",
    "prettier": "2.0.5",
    "regenerator-runtime": "^0.13.2"
  },
  "gitHead": "7e686ca8bfee944da876b6cfd2c32e71da244537",
  "name": "@launchql/utils",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "lql package",
    "test": "FAST_TEST=1 launchql-templatedb && jest",
    "test:watch": "FAST_TEST=1 jest --watch"
  },
  "version": "0.4.6"
}
