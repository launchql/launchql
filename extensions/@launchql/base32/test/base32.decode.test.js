import { getConnections } from './utils';
import cases from 'jest-in-case';

let db, base32, teardown;
const objs = {
  tables: {}
};

beforeAll(async () => {
  ({ db, teardown } = await getConnections());
  base32 = db.helper('base32');
});

afterAll(async () => {
  try {
    //try catch here allows us to see the sql parsing issues!
    await teardown();
  } catch (e) {
    // noop
  }
});

beforeEach(async () => {
  await db.beforeEach();
});

afterEach(async () => {
  await db.afterEach();
});

it('base32_to_decimal', async () => {
  const [{ base32_to_decimal }] = await base32.call('base32_to_decimal', {
    input: 'INQXI==='
  });
  expect(base32_to_decimal).toEqual([
    '8',
    '13',
    '16',
    '23',
    '8',
    '=',
    '=',
    '='
  ]);
});

it('base32_to_decimal', async () => {
  const [{ base32_to_decimal }] = await base32.call('base32_to_decimal', {
    input: 'INQXI==='
  });
  expect(base32_to_decimal).toEqual([
    '8',
    '13',
    '16',
    '23',
    '8',
    '=',
    '=',
    '='
  ]);
});

it('decimal_to_chunks', async () => {
  const [{ decimal_to_chunks }] = await base32.call(
    'decimal_to_chunks',
    {
      input: ['8', '13', '16', '23', '8', '=', '=', '=']
    },
    {
      input: 'text[]'
    }
  );
  expect(decimal_to_chunks).toEqual([
    '01000',
    '01101',
    '10000',
    '10111',
    '01000',
    'xxxxx',
    'xxxxx',
    'xxxxx'
  ]);
});

it('decode', async () => {
  const [{ decode }] = await base32.call('decode', {
    input: 'INQXI'
  });
  expect(decode).toEqual('Cat');
});

it('zero_fill', async () => {
  const [{ zero_fill }] = await base32.call('zero_fill', {
    a: 300,
    b: 2
  });
  expect(zero_fill).toBe('75');
});

it('zero_fill (-)', async () => {
  const [{ zero_fill }] = await base32.call('zero_fill', {
    a: -300,
    b: 2
  });
  expect(zero_fill).toBe('1073741749');
});

it('zero_fill (0)', async () => {
  const [{ zero_fill }] = await base32.call('zero_fill', {
    a: -300,
    b: 0
  });
  expect(zero_fill).toBe('4294966996');
});

cases(
  'base32',
  async (opts) => {
    const [result] = await base32.call('decode', {
      input: opts.name
    });
    expect(result.decode).toEqual(opts.result);
    expect(result.decode).toMatchSnapshot();
  },
  [
    { result: '', name: '' },
    { result: 'Cat', name: 'INQXI' },
    { result: 'chemistryisgreat', name: 'MNUGK3LJON2HE6LJONTXEZLBOQ======' },
    { result: 'f', name: 'MY======' },
    { result: 'fo', name: 'MZXQ====' },
    { result: 'foo', name: 'MZXW6===' },
    { result: 'foob', name: 'MZXW6YQ=' },
    { result: 'fooba', name: 'MZXW6YTB' },
    { result: 'fooba', name: 'mzxw6ytb' },
    { result: 'foobar', name: 'MZXW6YTBOI======' }
  ]
);
