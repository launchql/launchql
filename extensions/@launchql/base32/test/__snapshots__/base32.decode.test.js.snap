// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base32 INQXI 1`] = `"Cat"`;

exports[`base32 MNUGK3LJON2HE6LJONTXEZLBOQ====== 1`] = `"chemistryisgreat"`;

exports[`base32 MY====== 1`] = `"f"`;

exports[`base32 MZXQ==== 1`] = `"fo"`;

exports[`base32 MZXW6=== 1`] = `"foo"`;

exports[`base32 MZXW6YQ= 1`] = `"foob"`;

exports[`base32 MZXW6YTB 1`] = `"fooba"`;

exports[`base32 MZXW6YTBOI====== 1`] = `"foobar"`;

exports[`base32 case: 1 1`] = `""`;

exports[`base32 mzxw6ytb 1`] = `"fooba"`;
